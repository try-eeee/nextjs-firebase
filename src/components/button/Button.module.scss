.button {
  color: white;
  border: none;
  border-radius: 4px;
  font-weight: normal;
  cursor: pointer;
  transition: background-color 0.2s;
}

.default {
  background-color: #0070f3;

  &:hover {
    background-color: #005bb5;
  }
}

.danger {
  background-color: #d9003a;

  &:hover {
    background-color: #840000;
  }
}

.size-S {
  font-size: 0.875rem;
  padding: 0.35rem 0.75rem;
}

.size-M {
  font-size: 1rem;
  padding: 0.75rem 1.5rem;
}
