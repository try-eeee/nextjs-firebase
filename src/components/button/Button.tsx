"use client";

import styles from "./Button.module.scss";

type BUTTON_TYPES = "default" | "danger";
type BUTTON_SIZE = "size-S" | "size-M" | "size-L";

interface InputProps extends React.ComponentProps<"button"> {
  buttonType?: BUTTON_TYPES;
  buttonSize?: BUTTON_SIZE;
}

export function Button(props: InputProps) {
  const { buttonType = "default", buttonSize = "size-S", ...rest } = props;
  const type = `${styles[buttonType]}`;
  const size = `${styles[buttonSize]}`;

  return <button {...rest} className={`${styles.button} ${type} ${size}`} />;
}
