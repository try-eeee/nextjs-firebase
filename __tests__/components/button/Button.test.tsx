import React from "react";
import { render } from "@testing-library/react";
import { Button } from "@/components/button/Button";

describe("Button UnitTest", () => {

  test("Snapshot Test Of Button", () => {
    const { asFragment } = render(<Button />);
    expect(asFragment()).toMatchSnapshot();
  });
});
