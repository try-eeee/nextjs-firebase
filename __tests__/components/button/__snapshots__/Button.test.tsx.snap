// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button UnitTest Snapshot Test Of Button 1`] = `
<DocumentFragment>
  <button
    class="button default size-S"
  />
</DocumentFragment>
`;
